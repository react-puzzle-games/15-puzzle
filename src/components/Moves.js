import React, { Component, PropTypes } from 'react';
import styled from 'styled-components';

class Moves extends Component {
  static propTypes = {
    moves: PropTypes.number.isRequired,
  };

  render() {
    const { className, moves } = this.props;

    return (
      <div className={className}>
        Moves counter: {moves}
      </div>
    );
  }
}

export default styled(Moves)`
  margin: 5px;
`;
