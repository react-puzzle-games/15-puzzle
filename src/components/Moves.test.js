import renderer from 'react-test-renderer';
import React from 'react';
import Moves from './Moves';

describe('Moves renders correctly', () => {
  test('before start', () => {
    const tree = renderer.create(<Moves moves={0} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
