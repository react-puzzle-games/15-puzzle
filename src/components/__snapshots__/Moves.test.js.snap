exports[`Moves renders correctly before start 1`] = `
<div
  className="jbeTpO">
  Moves counter: 
  0
</div>
`;
